core/scheduler/service/replay_worker.go

```diff
@@ -105,7 +142,11 @@ func (w ReplayWorker) processNewReplayRequestParallel(ctx context.Context, repla
 }
 
 func (w ReplayWorker) processNewReplayRequestSequential(ctx context.Context, replayReq *scheduler.ReplayWithRun, jobCron *cron.ScheduleSpec) ([]*scheduler.JobRunStatus, error) {
-	runToClear := replayReq.GetFirstExecutableRun()
+	runsToBeCleared := scheduler.JobRunStatusList(replayReq.Runs).GetSortedRunsByStates([]scheduler.State{scheduler.StatePending})
```

Comment:
@deryrahman now that you have updated `GetFirstExecutableRun`, I believe you can use it here.